README.md

```diff
@@ -0,0 +1,713 @@
+
+* Regular expression flags
+
+    ```js
+    var diacritics = require("diacritics-transliterator");
+    var regexp = diacritics.createRegExp("Tést", {
+        flags: "giu"
+    });
+    // regexp => /T(\u00E9|e\u0301)st/giu
+    ```
+
+* Each callback
+
+    ```js
+    var diacritics = require("diacritics-transliterator");
+    var regexp = diacritics.createRegExp("a tést", {
+        each: function(character, result, data, index) {
+            // Replace space with regular expression whitespace
+            // make sure to double escape the RegExp special characters "\\s"
+            return character === " " ? "\\s+" : result;
+        }
+    });
+    // regexp => /a\s+t(\u00E9|e\u0301)st/gu
+    ```
+
+* Done callback
+
+    ```js
+    var diacritics = require("diacritics-transliterator");
+    var regexp = diacritics.createRegExp("tést", {
+        done: function(array, joiner) {
+            // match whole words only
+            // make sure to double escape the RegExp special characters "\\b"
+            return "\\b" + array.join(joiner) + "\\b";
+        }
+    });
+    // regexp => /\bt(\u00E9|e\u0301)st\b/gu
+    ```
+
+If no diacritics are found, then the regular expression will include the original text.
+
+#### __Important Notes__
+
+* This method is not meant to be used on HTML, and will escape any regular expression-like characters (e.g. `^(test)$` string is converted into the following string `\^\(test\)\$`).
+* If you want to include the start `^` and end `$` anchors, use the `done` callback in a manner similar to the example above.
+* Any regular expression special characters added within the callback functions must be double escaped (e.g. `\S` becomes `\\S`) because the regular expression is created from a string.
+* Adding an escaped unicode character within the callbacks must not be double escaped (e.g. `\u00a0` contains a single backslash). If you have included any double escaped unicode as found in the database, use the [`diacritics.formatUnicode` method](#diacriticsformatunicodestring) to process it.
+
+### diacritics.replacePlaceholder(string [, options])
+
+Replaces placeholder(s) within the string with the targeted diacritic values:
+- The `string` parameter (type: `string`)  contains text and/or HTML with a diacritic placeholder value(s) to be replaced.
+  - The query string contains two parts: `type;data`
+  - The `type` will match the diacritics.io database queries (e.g. `base=o`)
+  - The `data` will target the data to include (e.g. `alphabet`, `native`, `base`, `decompose` or `equivalents.unicode`) - including `metadata` or `data` is not required.
+  - Example: `<% diacritics: base=o;equivalents.unicode %>` will be replaced with `\u00FC,u\u0308,\u00FA,u\u0301` - this example is only showing the results from `de` and `es` languages; there will be a lot more once there is more data. The result can be reformatted using the `output` callback function.
```

Comment:
What are valid values other than `equivalents.unicode`? For example how to replace it with an array like [this](https://github.com/julmot/mark.js/blob/master/src/mark.js#L209)?